### PR TITLE
Adds 2FA support to git-fork

### PR DESCRIPTION
**Pull request by techjacker**
Two factor authentication support + minor linting

### CODE REVIEW

**Comment by hemanth** [comment]
LGTM.

File: bin/git-fork [hunk 1]
```diff
@@ -31,28 +31,27 @@ fi
...
+curl_args="-qs -X POST -u $user"
+curl_url="https://api.github.com/repos/$owner/$project/forks"
+RES=$(curl -H "X-GitHub-OTP: $MFA_CODE" $curl_args "$curl_url")
 [ $? = 0 ] || abort "fork failed"
 
 # Check if user has ssh configured with GitHub
-ssh -o BatchMode=yes git@github.com exit >/dev/null 2>&1
-ssh_exit_code=$?
-if [[ "$ssh_exit_code" = 0 ]] || [[ "$ssh_exit_code" = 1 ]]; then
+if ssh -T git@github.com 2>&1 | grep -qi 'success'; then
```
Comment (techjacker) [inline bin/git-fork]: It [seems possible](http://stackoverflow.com/questions/3663895/ssh-the-authenticity-of-host-hostname-cant-be-established). However doing so would expose the user to MITM attacks.

Comment (spacewander) [inline bin/git-fork]: Maybe we could tell users what will happen before `ssh`?
Just don't let them surprised.

Comment (techjacker) [inline bin/git-fork]: Sure, something like `echo 'Add github remote as SSH (you will be prompted to verify the server's credentials)?'`

Comment (spacewander) [inline bin/git-fork]: I think `Add GitHub remote branch via SSH (you will be prompted to verify the server's credentials)?` could be better.

Comment (techjacker) [inline bin/git-fork]: I added and rebased

**Comment by spacewander** [comment]
Merged.
Thanks, @techjacker

**Comment by techjacker** [comment]
You're welcome :)

**Pull request merged**